`coq-core.toplevel' not found in `make unit-tests`

<!-- Thank you for reporting a bug to Coq! -->

#### Description of the problem

<!-- If you can, it's helpful to provide self-contained example of some code
that reproduces the bug. If not, a link to a larger example is also helpful.
You can generate a shorter version of your program by following these
instructions: https://github.com/coq/coq/wiki/Coqbot-minimize-feature.  -->

```
make -f Makefile.dune world
cd test-suite/
make unit-tests
```
gives
```
ocamlfind: Package `coq-core.toplevel' not found
Makefile:325: recipe for target 'unit-tests/clib/inteq.ml.log' failed
make: *** [unit-tests/clib/inteq.ml.log] Error 2
```

#### Coq Version
8.15+alpha
<!-- You can get this information by running `coqtop -v`. If relevant, please
also include your operating system. -->

--- a/test-suite/Makefile
+++ b/test-suite/Makefile
@@ -29,6 +29,13 @@ COQ_SRC_DIR:=..
 
 -include ../config/Makefile
 
+ifneq ($(wildcard ../_build/default/config/Makefile),)
+include ../_build/default/config/Makefile
+ifeq ($(COQPREFIX),local)
+COQPREFIX:=$(shell cd ../_build/install/default/ ;pwd)
+endif
+endif
+
 #######################################################################
 # Variables
 #######################################################################
@@ -47,16 +54,14 @@ else
 endif
 
 # Be careful to use COQPREFIX when prefix is not set
+# this assumes we're running inside dune, ie pwd=_build/default/test-suite
+# is this assumption correct?
 ifeq ($(COQPREFIX),local)
 	COQPREFIX:=$(shell cd ../..;pwd)/install/default/
 endif
 
 export OCAMLPATH := $(shell echo $(COQPREFIX)/lib$(FINDLIB_SEP)$$OCAMLPATH)
 export CAML_LD_LIBRARY_PATH:=$(COQPREFIX)/lib/stublibs:$(CAML_LD_LIBRARY_PATH)
-
-ifneq ($(wildcard ../_build/default/config/Makefile),)
-include ../_build/default/config/Makefile
-endif
 
 # COQLIB is an env variable so no quotes
 COQLIB?=

